src/overlay/Peer.cpp

```diff
@@ -1905,13 +1880,14 @@ Peer::recvAuth(StellarMessage const& msg)
     // the other peer about the local node's reading capacity.
     sendSendMore(mApp.getConfig().PEER_FLOOD_READING_CAPACITY);
 
-    if (mRemoteOverlayVersion >= Peer::FIRST_VERSION_SUPPORTING_PULL_MODE &&
-        mApp.getConfig().OVERLAY_PROTOCOL_VERSION >=
-            Peer::FIRST_VERSION_SUPPORTING_PULL_MODE &&
-        msg.auth().flags == AUTH_MSG_FLAG_PULL_MODE_REQUESTED &&
-        mApp.getConfig().ENABLE_PULL_MODE)
+    if (mRemoteOverlayVersion < Peer::FIRST_VERSION_REQUIRING_PULL_MODE &&
+        msg.auth().flags != AUTH_MSG_FLAG_PULL_MODE_REQUESTED)
```

Comment:
this check doesn't seem right: version 27 still sets msg.auth().flags == AUTH_MSG_FLAG_PULL_MODE_REQUESTED

Reply:
I want Core to eventually stop using `AUTH_MSG_FLAG_PULL_MODE_REQUESTED` https://github.com/stellar/stellar-core/issues/3636 once every node's overlay version is >= 27.

I decided to go with this logic since:
- Case 1: peer's overlay version < 27
    - The peer may have turned off pull mode. We'll know that by checking the flag.
- Case 2: peer's overlay version >=.
    -  The peer definitely wants to use pull mode. The peer may have used `AUTH_MSG_FLAG_PULL_MODE_REQUESTED`, but we don't have to check it.

Reply:
I just realized that we know the peer's overlay version before we get to this line. (I mistakenly thought that that wasn't the case before 🤔 )

I updated the code so that we set `AUTH_MSG_FLAG_PULL_MODE_REQUESTED` only if the peer's overlay version is `< 27 = FIRST_VERSION_REQUIRING_PULL_MODE`.

Reply:
okay, that makes sense